runtime/service.go

```diff
@@ -1183,11 +1192,21 @@ func (s *service) shutdown(
 		s.shutdownLoop(requestCtx, timeout, req)
 	}()
 
-	err := s.machine.Wait(context.Background())
-	if err == nil {
+	var result *multierror.Error
+	if err := s.machine.Wait(context.Background()); err != nil {
+		result = multierror.Append(result, err)
+	}
+	if err := s.cleanup(); err != nil {
```

Comment:
Is the jailer.Close() here racing with the Jailer.Stop() from ShutdownLoop? Is there an expected order for them to be called in?

Reply:
s.machine.Wait() will return once Firecracker is terminated. cleanup() won't be called while shutdownLoop() is running.

Reply:
Good call. I missed that.